# Load in data 
+ Data comes as a tar file
+ I opted to just load in files into pandas instead of attempting to unwrap that tar file in python
+ We have a file containing all of the edges facebook.csv
+ What we need to do is combine the .feat files with the .featnames files
+ below you can see what each of these files looks like

In [216]:
import pandas as pd 

In [217]:



example= pd.read_csv('0.feat',sep=" ",header=None)
print(example.shape)
example.head()


(347, 225)


0    1    2    3    4    5    6    7    8    9   ...   215  216  217  218  \
0    1    0    0    0    0    0    0    0    0    0 ...     0    0    0    0   
1    2    0    0    0    0    0    0    0    0    0 ...     0    0    0    0   
2    3    0    0    0    0    0    0    0    1    0 ...     0    0    0    1   
3    4    0    0    0    0    0    0    0    0    0 ...     0    0    0    0   
4    5    0    0    0    0    0    0    0    0    0 ...     0    0    0    0   

   219  220  221  222  223  224  
0    0    0    0    0    0    0  
1    0    0    0    0    0    0  
2    0    0    0    0    0    0  
3    0    0    0    0    0    0  
4    0    0    0    0    0    0  

[5 rows x 225 columns]

In [218]:

example= pd.read_csv('0.featnames',sep=" ",header=None)
print(example.shape)
example.head()

(224, 4)


0                    1        2  3
0  0  birthday;anonymized  feature  0
1  1  birthday;anonymized  feature  1
2  2  birthday;anonymized  feature  2
3  3  birthday;anonymized  feature  3
4  4  birthday;anonymized  feature  4

# structure of our data

+  Column values in the .feat files(225 columns) are the equivalent index value for our .featnames files(224 rows)
    + The difference of 1 in the shape is due to the 1st column in the .feat file representing our node value and not a feature value
+ First goal is to join the names from .featnames in the feat files


### Build names lists
+ our files are stored as a random int.name
    + I.E  '0.featnames' and '0.feat'
     + Below we use list comprehension to build out file names


In [219]:
names=["0","107","1684","1912","3437",'348',"3980","414","686","698"]
feat_file_names= [x+".feat" for x in names]
featnames_filenames=[x+".featnames" for x in names]

In [220]:
# check names
featnames_filenames

['0.featnames',
 '107.featnames',
 '1684.featnames',
 '1912.featnames',
 '3437.featnames',
 '348.featnames',
 '3980.featnames',
 '414.featnames',
 '686.featnames',
 '698.featnames']

### Load files into pandas 

In [221]:
list_feat_names=[]
list_features=[]
for i,x in enumerate(featnames_filenames):
    list_feat_names.append(pd.read_csv(x,sep=" ",header=None))
    list_features.append(pd.read_csv(feat_file_names[i],sep=" ",header=None))

### Check shape matches earlier shape 

In [222]:
list_feat_names[0].shape

(224, 4)

In [223]:
list_features[0].shape

(347, 225)

### Practice on one set of each document

In [224]:
df=list_features[0]
df.head()
#df = df.drop(df.columns[[0, 1]], axis=1)
indexes=df.iloc[:,0]
df=df.iloc[:,1:]
df.columns= list_feat_names[0][1]
df["node"]=indexes
df.head()




1  birthday;anonymized  birthday;anonymized  birthday;anonymized  \
0                    0                    0                    0   
1                    0                    0                    0   
2                    0                    0                    0   
3                    0                    0                    0   
4                    0                    0                    0   

1  birthday;anonymized  birthday;anonymized  birthday;anonymized  \
0                    0                    0                    0   
1                    0                    0                    0   
2                    0                    0                    0   
3                    0                    0                    0   
4                    0                    0                    0   

1  birthday;anonymized  birthday;anonymized  education;classes;id;anonymized  \
0                    0                    0                                0   
1                    0                    0                                0   
2                    0                    1                                0   
3                    0                    0                                0   
4                    0                    0                                0   

1  education;classes;id;anonymized  ...   work;start_date;anonymized  \
0                                0  ...                            0   
1                                0  ...                            0   
2                                0  ...                            0   
3                                0  ...                            0   
4                                0  ...                            0   

1  work;start_date;anonymized  work;start_date;anonymized  \
0                           0                           0   
1                           0                           0   
2                           0                           1   
3                           0                           0   
4                           0                           0   

1  work;start_date;anonymized  work;start_date;anonymized  \
0                           0                           0   
1                           0                           0   
2                           0                           0   
3                           0                           0   
4                           0                           0   

1  work;start_date;anonymized  work;start_date;anonymized  \
0                           0                           0   
1                           0                           0   
2                           0                           0   
3                           0                           0   
4                           0                           0   

1  work;start_date;anonymized  work;with;id;anonymized  node  
0                           0                        0     1  
1                           0                        0     2  
2                           0                        0     3  
3                           0                        0     4  
4                           0                        0     5  

[5 rows x 225 columns]

In [227]:
final_product=[]
for i,x in enumerate(list_features):
    df=x
    indexes=x.iloc[:,0]
    df=x.iloc[:,1:]
    df.columns= list_feat_names[i][1]
    df["nodes"]=indexes
    final_prdocut.append(df)

In [228]:

final_prdocut[0].head()

1  birthday;anonymized  birthday;anonymized  birthday;anonymized  \
0                    0                    0                    0   
1                    0                    0                    0   
2                    0                    0                    0   
3                    0                    0                    0   
4                    0                    0                    0   

1  birthday;anonymized  birthday;anonymized  birthday;anonymized  \
0                    0                    0                    0   
1                    0                    0                    0   
2                    0                    0                    0   
3                    0                    0                    0   
4                    0                    0                    0   

1  birthday;anonymized  birthday;anonymized  education;classes;id;anonymized  \
0                    0                    0                                0   
1                    0                    0                                0   
2                    0                    1                                0   
3                    0                    0                                0   
4                    0                    0                                0   

1  education;classes;id;anonymized  ...    work;start_date;anonymized  \
0                                0  ...                             0   
1                                0  ...                             0   
2                                0  ...                             0   
3                                0  ...                             0   
4                                0  ...                             0   

1  work;start_date;anonymized  work;start_date;anonymized  \
0                           0                           0   
1                           0                           0   
2                           0                           1   
3                           0                           0   
4                           0                           0   

1  work;start_date;anonymized  work;start_date;anonymized  \
0                           0                           0   
1                           0                           0   
2                           0                           0   
3                           0                           0   
4                           0                           0   

1  work;start_date;anonymized  work;start_date;anonymized  \
0                           0                           0   
1                           0                           0   
2                           0                           0   
3                           0                           0   
4                           0                           0   

1  work;start_date;anonymized  work;with;id;anonymized  nodes  
0                           0                        0      1  
1                           0                        0      2  
2                           0                        0      3  
3                           0                        0      4  
4                           0                        0      5  

[5 rows x 225 columns]